crates/types/src/traits/block_contents.rs

```diff
@@ -79,22 +81,28 @@ pub trait BlockPayload:
     ///
     /// # Errors
     /// If the transaction length conversion fails.
+
     fn from_transactions(
         transactions: impl IntoIterator<Item = Self::Transaction>,
+        validated_state: &Self::ValidatedState,
         instance_state: &Self::Instance,
-    ) -> Result<(Self, Self::Metadata), Self::Error>;
+    ) -> impl Future<Output = Result<(Self, Self::Metadata), Self::Error>> + Send + 'static;
```

Comment:
Can you please add to this doc comment explaining why the async behavior is needed? From the trait, it's not abundantly clear. 

Could we perhaps just upgrade this trait to `async_trait` since we use it elsewhere in the code anyway?

Reply:
done in dec9dd6de58893f9080dde0b96285d05b2f64038